Mathlib/RingTheory/AlgebraicIndependent/Basic.lean

```diff
@@ -0,0 +1,400 @@
+/-
+Copyright (c) 2021 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import Mathlib.Algebra.MvPolynomial.Equiv
+import Mathlib.Algebra.MvPolynomial.Monad
+import Mathlib.Algebra.MvPolynomial.Supported
+import Mathlib.RingTheory.AlgebraicIndependent.Defs
+import Mathlib.RingTheory.Ideal.Maps
+import Mathlib.RingTheory.MvPolynomial.Basic
+
+/-!
+# Algebraic Independence
+
+This file contains basic results on algebraic independence of a family of elements of an `R`-algebra
+
+## References
+
+* [Stacks: Transcendence](https://stacks.math.columbia.edu/tag/030D)
+
+## TODO
+Define the transcendence degree and show it is independent of the choice of a
+transcendence basis.
+
+## Tags
+transcendence basis, transcendence degree, transcendence
+
+-/
+
+
+noncomputable section
+
+open Function Set Subalgebra MvPolynomial Algebra
+
+open scoped Classical
+
+universe u v w
```

Comment:
It would be good to check which of these are actually used in each file. I think `u v w` are not used in this file.